src/commands/createFunctionApp/FunctionAppCreateStep.ts

```diff
@@ -63,8 +62,40 @@ export class FunctionAppCreateStep extends AzureWizardExecuteStepWithActivityOut
         return !context.site;
     }
 
-    private async getNewSite(context: IFunctionAppWizardContext, stack: FullFunctionAppStack): Promise<Site> {
-        const site: Site = await this.createNewSite(context, stack);
+    async createFunctionApp(context: IFlexFunctionAppWizardContext, rgName: string, siteName: string, stack: FullFunctionAppStack): Promise<Site> {
+        let site: Site;
+        if (context.newSiteDomainNameLabelScope === DomainNameLabelScope.Global) {
+            site = await this.createSite(context, rgName, siteName, stack);
+        } else {
+            site = await this.createSiteWithDomainLabelScope(context, rgName, siteName, stack);
+        }
+
+        if (context.newFlexSku) {
+            if (context.storageAccount) {
+                await tryCreateStorageContainer(context, site, context.storageAccount);
+            }
+        }
+
+        return site;
+    }
+
+    // #region createSite
+    private async createSite(context: IFlexFunctionAppWizardContext, rgName: string, siteName: string, stack?: FullFunctionAppStack): Promise<Site> {
+        const client: WebSiteManagementClient = await createWebSiteClient(context);
+        const site: Site = context.newFlexSku ?
+            await this.getNewFlexSite(context, context.newFlexSku) :
+            await this.getNewSite(context, stack);
+        return await client.webApps.beginCreateOrUpdateAndWait(rgName, siteName, site)
+    }
+
+    private async getNewFlexSite(context: IFlexFunctionAppWizardContext, sku: Sku): Promise<Site> {
+        const site: Site = await this.getNewBaseSite(context);
+        site.functionAppConfig = this.getNewFlexFunctionAppConfig(context, sku);
+        return site;
+    }
+
+    private async getNewSite(context: IFunctionAppWizardContext, stack?: FullFunctionAppStack): Promise<Site> {
```

Comment:
Some of the diff here is showing up as new code, but much of it is just me reorganizing existing code sections and grouping related logic under their own foldable regions.  

Reorganizing was especially important because we temporarily duplicated (doubled) the number of branching logic paths due to needing to make a manual call without the SDK (which requires forming different shaped site payloads).